python/lsst/ts/m2gui/controltab/tab_utility_view.py

```diff
@@ -33,7 +39,424 @@ class TabUtilityView(TabDefault):
+            )
+
+            for sensor in sensors:
+                layout.addRow(sensor + ":", self._temperatures[sensor])
+
+            self._add_empty_row_to_layout(layout)
+
+        return create_group_box(group_title, layout)
+
+    def _add_empty_row_to_layout(self, layout):
+        """Add the empty row to the layout.
+
+        Parameters
+        ----------
+        layout : `PySide2.QtWidgets.QFormLayout`
+            Layout.
+        """
+        layout.addRow("", create_label())
+
+    def _create_group_displacements(self):
+        """Create the group of displacement sensors.
+
+        Returns
+        -------
+        group : `PySide2.QtWidgets.QGroupBox`
+            Group.
+        """
+
+        layout = QFormLayout()
+
+        sensors_theta = self.model.utility_monitor.get_displacement_sensors(
+            DisplacementSensorDirection.Theta
+        )
+        sensors_delta = self.model.utility_monitor.get_displacement_sensors(
+            DisplacementSensorDirection.Delta
+        )
+
+        for sensor_theta, sensor_delta in zip(sensors_theta, sensors_delta):
+            layout.addRow(sensor_theta + ":", self._displacements[sensor_theta])
+            layout.addRow(sensor_delta + ":", self._displacements[sensor_delta])
+            self._add_empty_row_to_layout(layout)
+
+        return create_group_box("Displacement Sensors", layout)
+
+    def _set_signal_utility(self, signal_utility):
+        """Set the utility signal with callback functions. This signal provides
+        the utility status contains the power, inclinometer, breaker,
+        temperature, displacements, etc.
+
+        Parameters
+        ----------
+        signal_utility : `SignalUtility`
+            Signal of the utility.
+        """
+
+        signal_utility.power_motor.connect(self._callback_power_motor)
+        signal_utility.power_communication.connect(self._callback_power_communication)
+
+        signal_utility.inclinometer.connect(self._callback_inclinometer)
+
+        signal_utility.breaker_status.connect(self._callback_breakers)
+
+        signal_utility.temperatures.connect(self._callback_temperatures)
+
+        signal_utility.displacements.connect(self._callback_displacements)
+
+    @Slot()
+    def _callback_power_motor(self, power_motor):
+        """Callback of the utility signal for the motor power.
+
+        Parameters
+        ----------
+        power_motor : `tuple`
+            Motor power: (voltage, current). The data type is float. The units
+            are volt and ampere respectively.
+        """
+
+        self._power_inclinometer["power_voltage_motor"].setText(f"{power_motor[0]} V")
+        self._power_inclinometer["power_current_motor"].setText(f"{power_motor[1]} A")
+
+    @Slot()
+    def _callback_power_communication(self, power_communication):
+        """Callback of the utility signal for the communication power.
+
+        Parameters
+        ----------
+        power_communication : `tuple`
```

Comment:
Passing tuple is dangerous, you can easily change 0/1 and display/use wrong values. dict/map from utility monitor can be easily passed instead, yet even better you can create class which will hold fields (even a generic class, using __get_attr__ calls to mimic what it received from json/yaml? - https://stackoverflow.com/questions/6578986/how-to-convert-json-data-into-a-python-object). Then you will pass resulting class, and in following, use setText(power_communication.voltage) etc. This will make code easily extendible (if needed), safer, and easier to read/verify.

Reply:
This is good idea for the C++ as a strong-typed language. Unfortunately, the Python is a weak-typed language. Therefore, you can pass whatever class/object you want (and the `tuple` is an object as well). You can see I have the enough documents and comments to explain the details. All functions in the class will `emit` the details by an interface function as well. I also use the unit tests to check all the signal details. I think this protection should be enough.

In addition, I have the bad experience in the previous project that people have too many classes to complex the code readability and maintenance. Therefore, I prefer to only add the new classes when needed. Thanks!